lib/hooksData.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const rp = require('request-promise-native')
+
+class HooksData {
+
+    constructor(taskId, user, pass) {
+        this.taskId = taskId;
+        this.user = user;
+        this.pass = pass;
+        this.basePath = process.env.ELASTICIO_API_URI || 'https://api.elastic.io/';
+    }
+
+    request(method, data) {
```

Comment:
If ``data`` is not defined or is an empty json object, we can print a warning a return here immdiatelly